File: java/168.excel表列名称.java
/*
 * @lc app=leetcode.cn id=168 lang=java
 *
 * [168] Excel表列名称
 */

// @lc code=start
// 使用进制转换的方法
class Solution {
    public String convertToTitle(int n) {
        String c[] = {"", "A", "B", "C", "D", "E", "F", "G", "H", "I","J", "K", "L","M" ,"N", "O", "P", "Q","R", "S", "T","U","V", "W","X", "Y","Z"};
        String res = "";
        while (n > 26) {
            if (n % 26 == 0) {
               res = "Z" + res;
               n =  n / 26 - 1;
            } else {
                res = c[n % 26 ]  + res;
                n = n / 26; 
            }
        }
        res = c[n] + res;
        return res;
    }
}
// @lc code=end

File: javascript/168.excel表列名称.js
/*
 * @lc app=leetcode.cn id=168 lang=javascript
 *
 * [168] Excel表列名称
 */

// @lc code=start
/**
 * @param {number} n
 * @return {string}
 */
// ASCII 转换的方法， 还可以通过进制转换 26进制，但是有一点区别
var convertToTitle = function(n) {
    // 返回值
    var res = "";
    while(n > 26) {
        var mod = n % 26;
        if (mod == 0) { // 最后一位为 Z
            res = res + "Z";
            n = n / 26 - 1;
        } else {
            res = String.fromCharCode(n % 26  + 48 + 16) + res;
            n = Math.floor(n / 26);
        };
    };
    if (n != 0) {
        res = String.fromCharCode(n + 48 + 16) + res; 
    };
    return res;
// @lc code=end

